conans/client/graph/graph.py

```diff
@@ -335,3 +327,46 @@ def _order_levels(self, direct):
             opened = opened.difference(current_level)
 
         return result
+
+    def private_skip_binaries(self, nodes_subset=None, root=None):
+        """ check which nodes are reachable from the root, mark the non reachable as BINARY_SKIP.
+        Used in the GraphBinaryAnalyzer"""
+        public_nodes = set()
+        root = root if root is not None else self.root
+        nodes = nodes_subset if nodes_subset is not None else self.nodes
+        current = [root]
+        while current:
+            new_current = set()
+            public_nodes.update(current)
+            for n in current:
+                if n.binary in (BINARY_CACHE, BINARY_DOWNLOAD, BINARY_UPDATE, BINARY_SKIP):
+                    # Might skip deps
+                    to_add = [d.dst for d in n.dependencies if not d.private]
+                else:
+                    # sure deps doesn't skip
+                    to_add = set(n.neighbors()).difference(public_nodes)
+                new_current.update(to_add)
+            current = new_current
+
+        for node in nodes:
+            if node not in public_nodes:
+                node.binary_non_skip = node.binary
+                node.binary = BINARY_SKIP
+
+    def build_time_nodes(self):
```

Comment:
I'm proposing a different name, something related to the fact that these nodes [usually/should] contain actual libraries that will be linked/deployed with the _root_ application: `library_nodes`? `library_host_nodes`? Not sure, but for me the name `build_time` is associated with our _build requires_ (nodes that should be there at the build time) 🤷‍♀

Reply:
Note that at the end of the method, it is computing the build-requires nodes, not the library ones

Reply:
Then... the docstring says the opposite: "return all the connected nodes to the root that are not build-requires..."